clang/lib/Driver/Driver.cpp

```diff
@@ -5203,99 +5225,558 @@ class OffloadingActionBuilder final {
       }
     }
 
-    // Return whether to use native bfloat16 library.
-    bool selectBfloatLibs(const ToolChain *TC, bool &useNative) {
-      const char *TargetOpt = nullptr;
-      const char *DeviceOpt = nullptr;
-      bool needLibs = false;
-      for (auto *A : Args) {
-        llvm::Triple *TargetBE = nullptr;
-
-        auto GetTripleIt = [&, this](llvm::StringRef Triple) {
-          llvm::Triple TargetTriple{Triple};
-          auto TripleIt = llvm::find_if(SYCLTripleList, [&](auto &SYCLTriple) {
-            return SYCLTriple == TargetTriple;
-          });
-          return TripleIt != SYCLTripleList.end() ? &*TripleIt : nullptr;
-        };
+    // Performs device specific linking steps for the SYCL based toolchain.
+    // This function is used for both the early AOT flow and the typical
+    // offload device link flow.
+    // The ListIndex input provides an index against the SYCLTargetInfoList
+    // used to determine associated toolchain information for the values being
+    // worked against to add the device link steps.  This is also used to change
+    // how the toolchain is formed, either by dependency additions or by adding
+    // actions explicitly.
+    void appendSYCLDeviceLink(ActionList &ListIndex, const ToolChain *TC,
+                              OffloadAction::DeviceDependences &DA,
```

Comment:
What is DA referring to here? DeviceArch? Dependency Addition?
I usually find comments that describe what the individual parameters are, to be helpful.
Just like here: https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/Flang.h#L30